indra/assemblers/html/assembler.py

```diff
@@ -50,18 +106,274 @@ def color_gen(scheme):
+                      force_reload: bool = False) -> SourceColors:
+    """Get the color scheme for a list of sources
+
+    Note: The input source names are assumed to be as they appear in INDRA,
+    i.e. source_info.json, but will be translated to the names used in
+    INDRA DB in the source colors returned.
+
+    Parameters
+    ----------
+    sources :
+        A list of source names as they appear in source_info.json
+    force_reload :
+        If True, reload source_info.json
+
+    Returns
+    -------
+    SourceColors
+        The source colors of the requested
+    """
+    # Pass the collected source info to _source_info_to_source_colors
+    source_info_json = _get_source_info(force_reload)
+    source_info = {}
+    for source in sources:
+        if source in SOURCE_INFO:
+            # source_info_json contains the drum-trips duplication, so trips
+            # is allowed as source
+            source_info[source] = source_info_json[source]
+        else:
+            logger.warning(f'Source {source} not recognized')
+
+    return _source_info_to_source_colors(source_info)
+
+
+def get_default_source_colors(force_reload: bool = False) -> SourceColors:
+    """Get the default color theme from source_info.json
+
+    Parameters
+    ----------
+    force_reload :
+        Forces a reload of source_info.json
+
+    Returns
+    -------
+    SourceColors
+        A source to color style mapping with naming of sources consistent
+        with the naming in INDRA DB
+    """
+    # Be aware of the 'hack' that lives in SOURCE_INFO but not in the
+    # actual file source_info.json:
+    # SOURCE_INFO['trips'] = SOURCE_INFO['drum']
+    # Also be aware of the INDRA - INDRA DB inconsistency in source api
+    # naming, the mapping of which lives in 'internal_source_mappings'
+
+    source_info_json = _get_source_info(force_reload)
+    # Copy info for pc into biopax, as INDRA DB treats them as the same and
+    # uses pc for both
+    source_info_copy = copy.deepcopy(source_info_json)
+    source_info_copy['biopax']['default_style'] = \
+        source_info_copy['pc']['default_style']
+
+    return _source_info_to_source_colors(source_info_json)
+
+
+try:
+    DEFAULT_SOURCE_COLORS = get_default_source_colors()
+except KeyError as err:
+    logger.info('source_info.json seems to have changed, regenerating...')
+    regenerate_default_source_styling()
+    DEFAULT_SOURCE_COLORS = get_default_source_colors(force_reload=True)
+
+
+def generate_source_css(fname: str,
+                        source_colors: SourceColors = None):
+    """Save a stylesheet defining color, background-color for the given sources
+
+    Parameters
+    ----------
+    fname :
+        Where to save the stylesheet
+    source_colors :
+        Colors defining the styles
+    """
+    if source_colors is None:
+        source_colors = DEFAULT_SOURCE_COLORS
+
+    rule_string = '.source-{src} {{\n    background-color: {src_bg};\n    ' \
```

Comment:
might be overkill, but you could use a jinja template for this

Reply:
I think that would re-open the rabbit hole that I was in in the previous attempt (#1297) :sweat_smile: